pkg/asset/manifests/dns.go

```diff
@@ -81,11 +80,15 @@ func (d *DNS) Generate(dependencies asset.Parents) error {
 	switch installConfig.Config.Platform.Name() {
 	case awstypes.Name:
 		if installConfig.Config.Publish == types.ExternalPublishingStrategy {
-			zone, err := icaws.GetPublicZone(installConfig.Config.BaseDomain)
-			if err != nil {
-				return errors.Wrapf(err, "getting public zone for %q", installConfig.Config.BaseDomain)
+			publicZoneID := installConfig.Config.AWS.PublicZoneID
+			if publicZoneID == "" {
+				zone, err := icaws.GetPublicZone(installConfig.Config.BaseDomain)
+				if err != nil {
+					return errors.Wrapf(err, "getting public zone for %q", installConfig.Config.BaseDomain)
+				}
+				publicZoneID = *zone.Id
 			}
-			config.Spec.PublicZone = &configv1.DNSZone{ID: strings.TrimPrefix(*zone.Id, "/hostedzone/")}
+			config.Spec.PublicZone = &configv1.DNSZone{ID: publicZoneID}
```

Comment:
hmm, should we change the `strings.TrimPrefix(*zone.Id, "/hostedzone/")` -> `publicZoneID` ? that might break the clients in the cluster?